vendor/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go

```diff
@@ -208,18 +222,28 @@ func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	o.nodeInfos = []*resource.Info{}
 	o.mapper, o.typer = o.Factory.Object()
 
 	cmdNamespace, _, err := o.Factory.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
+	nameArgs := []string{"nodes"}
+	if len(args) > 0 {
+		nameArgs = append(nameArgs, args[0])
+		if strings.Contains(args[0], "/") {
+			nameArgs = []string{args[0]}
+		}
+	}
+
 	r := o.Factory.NewBuilder(true).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
-		ResourceNames("node", args[0]).
+		SelectorParam(o.Selector).
```

Comment:
shoudn't this be `ResourceNames("nodes", args...)`?

Reply:
the `nameArgs := []string{"nodes"}` and `if strings.Contains(args[0], "/")` block above means this command parses resource args differently than other commands... 

I'd expect `ResourceNames("nodes", args...).SingleResourceType()` and a check on the resource type of selected resources to make sure they are nodes

Reply:
link to the upstream PR that resolved this? also, did 52917 go into 1.8?

Reply:
Upstream patch: https://github.com/kubernetes/kubernetes/pull/54083
52917 is not in 1.8: [https://github.com/kubernetes/kubernetes/blob/release-1.8/pkg/kubectl/cmd/drain.go]()